src/toil/worker.py

```diff
@@ -259,10 +259,6 @@ def badWorker():
                 else:
                     jobGraph.command = jobGraph.checkpoint
 
-                # Reduce the retry count
-                assert jobGraph.remainingRetryCount >= 0
-                jobGraph.remainingRetryCount = max(0, jobGraph.remainingRetryCount - 1)
```

Comment:
This part is what was causing the "failure" on a successful resume. The retry count was decremented during "cleanup" of a restarted checkpoint job--even if the checkpoint was restarted because the workflow was restarted. This is now handled by the `setupJobAfterFailure` line in the leader.